toolsrc/src/vcpkg/base/downloads.cpp

```diff
@@ -11,131 +15,161 @@
 namespace vcpkg::Downloads
 {
 #if defined(_WIN32)
-    static void winhttp_download_file(Files::Filesystem& fs,
-                                      ZStringView target_file_path,
-                                      StringView hostname,
-                                      StringView url_path)
+    struct WinHttpCloseHandleWrapper
     {
-        // Make sure the directories are present, otherwise fopen_s fails
-        const auto dir = fs::path(target_file_path.c_str()).parent_path();
-        std::error_code ec;
-        fs.create_directories(dir, ec);
-        Checks::check_exit(VCPKG_LINE_INFO, !ec, "Could not create directories %s", fs::u8string(dir));
-
-        FILE* f = nullptr;
-        const errno_t err = fopen_s(&f, target_file_path.c_str(), "wb");
-        Checks::check_exit(VCPKG_LINE_INFO,
-                           !err,
-                           "Could not download https://%s%s. Failed to open file %s. Error code was %s",
-                           hostname,
-                           url_path,
-                           target_file_path,
-                           std::to_string(err));
-        ASSUME(f != nullptr);
-
-        auto hSession = WinHttpOpen(L"vcpkg/1.0",
-                                    IsWindows8Point1OrGreater() ? WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY
-                                                                : WINHTTP_ACCESS_TYPE_DEFAULT_PROXY,
-                                    WINHTTP_NO_PROXY_NAME,
-                                    WINHTTP_NO_PROXY_BYPASS,
-                                    0);
-        Checks::check_exit(VCPKG_LINE_INFO, hSession, "WinHttpOpen() failed: %d", GetLastError());
-
-        // If the environment variable HTTPS_PROXY is set
-        // use that variable as proxy. This situation might exist when user is in a company network
-        // with restricted network/proxy settings
-        auto maybe_https_proxy_env = System::get_environment_variable("HTTPS_PROXY");
-        if (auto p_https_proxy = maybe_https_proxy_env.get())
+        void operator()(HINTERNET h) { WinHttpCloseHandle(h); }
```

Comment:
```suggestion
        void operator()(HINTERNET h) const { WinHttpCloseHandle(h); }
```
might as well be const correct.